hudi-client/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -113,6 +113,9 @@
   public static final String MAX_CONSISTENCY_CHECKS_PROP = "hoodie.consistency.check.max_checks";
   public static int DEFAULT_MAX_CONSISTENCY_CHECKS = 7;
 
+  private static final String PAYLOAD_ORDERING_FIELD_PROP = "hoodie.payload.ordering.field";
```

Comment:
@vinothchandar @bhasudha : can you throw some light on why new config rather than reusing preCombine field value?

Reply:
That's because you don't know what the ordering field value is, when reading the data from storage right?

Reply:
ok, so for preCombine (combining two records within the same ingestion batch) we use one config and for determining ordering among one incoming record and another in storage or in other words, to combine an incoming record w/ one in storage, we use a diff config? 
I see both these as one and the same. just that whether we resolve records within same batch or across batches is what differs.

Reply:
You are right, they are one and the same. If we can avoid the additional config, we should

Reply:
gotcha. here is my take. 
Will introduce a config called "honorOrderingToCombineRecordsAcrossBatches" (we can work out a good naming). default value is false. Will send in preCombineFieldKey as an arg to OverwriteWithLatestAvroPayload.

```
public OverwriteWithLatestAvroPayload(GenericRecord record, Comparable orderingVal, String precombineFieldKey, boolean honorOrderingToCombineRecordsAcrossBatches) {
    super(record, orderingVal);
.
```
Based on "honorOrderingToCombineRecordsAcrossBatches" value, combineAndGetUpdateValue() impl will decide to go with existing impl or new one (as we see in this patch) 

I know we do have a gap here where in, if ordering field has changed over time, we can't do much here. But we can't afford to store the ordering field as a separate column in dataset either, as diff commits could theoretically have diff ordering field. But guess we can call it out that we may not support such evolution of ordering field.

Reply:
I prefer we don't do this. but separate the existing impl and new one into separate payload classes. 

don't understand actually how we plan to implement obtaining the ordering field from the record in dfs. could you please clarify? @nsivabalan

Reply:
similar to how this patch already does. basically we lookup the preCombine field in generic record.

```
Object persistedOrderingVal = getNestedFieldVal((GenericRecord) currentValue, props.get(ORDERING_FIELD_OPT_KEY), true);
```
Reason I proposed to tweak existing class is. 
Even if we want to go w/ 2 classes, Ideally I would want to name existing class as OverwriteWithIncomingPayload and introduce a new class called OverwriteWithLatestAvroPayload, bcoz, existing one overwrites w/ incoming and we are introduce a new class which will actually overwrite w/ latest payload. 
But since users configure via class names, we can't go w/ this approach and hence my proposal to introduce a config and change existing class.